core/git_command.py

```diff
@@ -31,6 +31,22 @@ class GitSavvyError(Exception):
     pass
 
 
+class IsolatedGitCommand(GitCommand):
```

Comment:
Not sure this is the abstraction I'd like to go with, so will think on this a bit before moving forward.